android/app/src/main/java/app/organicmaps/sdk/sync/SyncPrefs.java

```diff
@@ -0,0 +1,407 @@
+  private final List<AccountToggledCallback> mAccountToggledCallbacks = new ArrayList<>(2);
+  private final List<FrequencyChangeCallback> mFrequencyChangeCallbacks = new ArrayList<>(2);
+
+  private SyncPrefs(Application context)
+  {
+    prefsAccounts = context.getSharedPreferences(PREF_NAME_ACCOUNTS, Context.MODE_PRIVATE);
+    Set<String> storedAccounts = prefsAccounts.getStringSet(PREF_KEY_ACCOUNTS, Collections.emptySet());
+    mAccounts = new CopyOnWriteArrayList<>(storedAccounts.stream()
+                                               .map(accountStr -> {
+                                                 try
+                                                 {
+                                                   return SyncAccount.fromJson(new JSONObject(accountStr));
+                                                 }
+                                                 catch (JSONException e)
+                                                 {
+                                                   Logger.e(TAG, "Invalid stored account JSON: " + accountStr, e);
+                                                   return null;
+                                                 }
+                                               })
+                                               .filter(Objects::nonNull)
+                                               .collect(Collectors.toList()));
+  }
+
+  public static SyncPrefs getInstance(Context context)
+  {
+    if (instance == null)
+      instance = new SyncPrefs((Application) context.getApplicationContext());
+    return instance;
+  }
+
+  private String getPrefKeyEnabled(long accountId)
+  {
+    return PREF_KEY_PREFIX_ENABLED + accountId;
+  }
+
+  private String getPrefKeyLastSynced(long accountId)
+  {
+    return PREF_KEY_PREFIX_LAST_SYNCED + accountId;
+  }
+
+  private String getPrefKeyErrorInfo(long accountId)
+  {
+    return PREF_KEY_PREFIX_ERROR_INFO + accountId;
+  }
+
+  public boolean isEnabled(long accountId)
+  {
+    return prefsAccounts.getBoolean(getPrefKeyEnabled(accountId), false);
+  }
+
+  public void setEnabled(SyncAccount account, boolean enabled)
+  {
+    prefsAccounts.edit().putBoolean(getPrefKeyEnabled(account.getAccountId()), enabled).apply();
+    for (AccountToggledCallback callback : mAccountToggledCallbacks)
+    {
+      try
+      {
+        if (enabled)
+          callback.onAccountEnabled(account);
+        else
+          callback.onAccountDisabled(account);
+      }
+      catch (Exception ignored)
+      {}
+    }
+  }
+
+  public List<SyncAccount> getAccounts()
+  {
+    return Collections.unmodifiableList(mAccounts);
+  }
+
+  public List<SyncAccount> getEnabledAccounts()
+  {
+    return mAccounts.stream()
+        .filter(account -> isEnabled(account.getAccountId()))
+        .collect(Collectors.toUnmodifiableList());
+  }
+
+  /**
+   * @return timestamp in milliseconds
+   */
+  public @Nullable Long getLastSynced(long accountId)
```

Comment:
```suggestion
  @Nullable
  public Long getLastSynced(long accountId)
```

And fix it in other places as well please